projects/dc2677a/common/dc2677a_qsys.tcl

```diff
@@ -0,0 +1,139 @@
+set_instance_parameter_value adc_pwm_gen {PULSE_0_WIDTH} {3}
+set_instance_parameter_value adc_pwm_gen {PULSE_0_PERIOD} {400}
+set_instance_parameter_value adc_pwm_gen {PULSE_0_OFFSET} {0}
+add_interface axi_ltc235x_cnv_if conduit end
+set_interface_property axi_ltc235x_cnv_if EXPORT_OF adc_pwm_gen.if_pwm_0
+add_connection sys_clk.clk adc_pwm_gen.if_ext_clk
+add_connection sys_clk.clk adc_pwm_gen.s_axi_clock
+add_connection sys_clk.clk_reset adc_pwm_gen.s_axi_reset
+
+# pack
+
+add_instance util_adc_pack util_cpack2
+set_instance_parameter_value util_adc_pack {NUM_OF_CHANNELS} $ADC_NUM_CHANNELS
+set_instance_parameter_value util_adc_pack {SAMPLES_PER_CHANNEL} {1}
+set_instance_parameter_value util_adc_pack {SAMPLE_DATA_WIDTH} {32}
+add_connection sys_clk.clk util_adc_pack.clk
+add_connection sys_clk.clk_reset util_adc_pack.reset
+for {set i 0} {$i < $ADC_NUM_CHANNELS} {incr i} {
+    add_connection axi_ltc235x.adc_ch_$i util_adc_pack.adc_ch_$i
+}
+add_connection util_adc_pack.if_fifo_wr_overflow axi_ltc235x.if_adc_dovf
+
+# dmac
+
+add_instance axi_adc_dma axi_dmac
+set_instance_parameter_value axi_adc_dma {ID} {0}
+if {$ADC_NUM_CHANNELS == 8} {
+    set_instance_parameter_value axi_adc_dma {DMA_DATA_WIDTH_SRC} {256}
+} elseif {$ADC_NUM_CHANNELS == 4} {
+    set_instance_parameter_value axi_adc_dma {DMA_DATA_WIDTH_SRC} {128}
+} else {
+    set_instance_parameter_value axi_adc_dma {DMA_DATA_WIDTH_SRC} {64}
+}
+set_instance_parameter_value axi_adc_dma {DMA_DATA_WIDTH_DEST} {64}
+set_instance_parameter_value axi_adc_dma {DMA_2D_TRANSFER} {0}
+set_instance_parameter_value axi_adc_dma {AXI_SLICE_DEST} {0}
+set_instance_parameter_value axi_adc_dma {AXI_SLICE_SRC} {0}
+set_instance_parameter_value axi_adc_dma {SYNC_TRANSFER_START} {1}
+set_instance_parameter_value axi_adc_dma {CYCLIC} {0}
+set_instance_parameter_value axi_adc_dma {DMA_TYPE_DEST} {0}
+set_instance_parameter_value axi_adc_dma {DMA_TYPE_SRC} {2}
+set_instance_parameter_value axi_adc_dma {FIFO_SIZE} {4}
+add_connection sys_clk.clk axi_adc_dma.s_axi_clock
+add_connection sys_clk.clk_reset axi_adc_dma.s_axi_reset
+add_connection sys_clk.clk axi_adc_dma.m_dest_axi_clock
+add_connection sys_clk.clk_reset axi_adc_dma.m_dest_axi_reset
+add_connection sys_clk.clk axi_adc_dma.if_fifo_wr_clk
+add_connection util_adc_pack.if_packed_fifo_wr_en axi_adc_dma.if_fifo_wr_en
+add_connection util_adc_pack.if_packed_fifo_wr_sync axi_adc_dma.if_fifo_wr_sync
+add_connection util_adc_pack.if_packed_fifo_wr_data axi_adc_dma.if_fifo_wr_din
+add_connection axi_adc_dma.if_fifo_wr_overflow util_adc_pack.if_packed_fifo_wr_overflow
+
+# interrupts / cpu interrupts
+
+ad_cpu_interrupt 2 axi_adc_dma.interrupt_sender
+
+# cpu interconnects / address map
+
+ad_cpu_interconnect 0x00120000 axi_ltc235x.s_axi
+ad_cpu_interconnect 0x00140000 adc_pwm_gen.s_axi
+ad_cpu_interconnect 0x00100000 axi_adc_dma.s_axi
+
+# mem interconnects / dma interconnects
+
+ad_dma_interconnect axi_adc_dma.m_dest_axi 1
+
```

Comment:
Remove the last empty line, but make sure you have a `\n` character at the end of the line, so the red sign will not appear.